opm/simulators/wells/BlackoilWellModelGeneric.cpp

```diff
@@ -91,20 +91,27 @@ BlackoilWellModelGeneric(Schedule& schedule,
     , nupcol_wgstate_(phase_usage)
 {
 
-  const auto numProcs = comm_.size();
-  this->not_on_process_ = [this, numProcs](const Well& well) {
-      if (numProcs == decltype(numProcs){1})
-          return false;
-
-      // Recall: false indicates NOT active!
-      const auto value = std::make_pair(well.name(), true);
-      auto candidate = std::lower_bound(this->parallel_well_info_.begin(),
-                                        this->parallel_well_info_.end(),
-                                        value);
-
-      return (candidate == this->parallel_well_info_.end())
-              || (*candidate != value);
-  };
+    const auto numProcs = comm_.size();
+    this->not_on_process_ = [this, numProcs](const Well& well) {
+        if (numProcs == decltype(numProcs){1})
+            return false;
+
+        // Recall: false indicates NOT active!
+        const auto value = std::make_pair(well.name(), true);
+        auto candidate = std::lower_bound(this->parallel_well_info_.begin(),
+                                          this->parallel_well_info_.end(),
+                                          value);
+
+        return (candidate == this->parallel_well_info_.end())
+                || (*candidate != value);
+    };
+
+    const auto& node_pressures = eclState.getRestartNetworkPressures();
+    if (node_pressures.has_value()) {
+        for (const auto& it: node_pressures.value()) {
```

Comment:
Can we do 
```c++
this->node_pressures_ = node_pressures.value();
```?

Reply:
Absolutely, that's better - will update. Thanks!

Reply:
It is more like a question.  If there are more complicated scenarios that we can not do assigning, we will know and deal with it.